core/driver/sources/src/test/scala/org/virtuslab/ideprobe/dependencies/IntelliJCleanupTest.scala

```diff
@@ -0,0 +1,76 @@
+package org.virtuslab.ideprobe.dependencies
+
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import org.virtuslab.ideprobe.config.{DependenciesConfig, IdeProbeConfig, IntelliJProvider, IntellijConfig, PathsConfig}
+import org.virtuslab.ideprobe.ide.intellij.IntelliJFactory
+
+import java.nio.file.Path
+
+@RunWith(classOf[JUnit4])
+final class IntelliJCleanupTest {
+  @Test
+  def existingIntelliJShouldNotBeDeletedDuringCleanup: Unit = givenInstalledIntelliJ { installationRoot =>
+      val intelliJProvider = IntelliJProvider.from(
+        IdeProbeConfig(
+          intellij = IntellijConfig.Existing(installationRoot, Seq.empty),
+          workspace = None,
+          resolvers = DependenciesConfig.Resolvers(),
+          driver = IntelliJFactory.Default.config,
+          paths = PathsConfig()
+        )
+      )
+      val existingInstalledIntelliJ = intelliJProvider.setup()
+
+      //when
+      existingInstalledIntelliJ.cleanup()
+
+      //then
+      assert(installationRoot.toFile.exists(), "The provided IntelliJ instance should exist after cleanup, but it was deleted.")
```

Comment:
either add exists to PathExtensions or use isFile() or isDirectory()

Reply:
Done.